### PR TITLE
Upgrade to ESLint 8

### PR DESCRIPTION
**Pull request by dlarocque**
Upgrade to ESLint 8. This fixes a peer dependency issue:
```
warning " > eslint-plugin-import@2.26.0" has incorrect peer dependency "eslint@^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8".
```

I tested this works by running `yarn lint`, and `yarn api-extractor`.

### CODE REVIEW

File: packages/analytics/src/helpers.ts [hunk 1]
```diff
@@ -185,6 +185,7 @@ async function gtagOnEvent(
     // if not all entries in the 'send_to' field could be mapped to
     // a FID. In these cases, wait on all pending initialization promises.
     if (initializationPromisesToWaitFor.length === 0) {
+      /* eslint-disable @typescript-eslint/no-floating-promises */
```
Comment (DellaBitta) [inline packages/analytics/src/helpers.ts:188]: This should have a corresponding eslint-enabled after the "errant" statement below.

File: packages/auth-compat/test/helpers/helpers.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 import * as sinon from 'sinon';
 import firebase from '@firebase/app-compat';
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (DellaBitta) [inline packages/auth-compat/test/helpers/helpers.ts:20]: This should have a corresponding eslint-enabled after the "errant" statement below.

File: packages/firestore/src/util/async_queue_impl.ts [hunk 3]
```diff
@@ -286,6 +286,7 @@ export class AsyncQueueImpl implements AsyncQueue {
     // Note that draining may generate more delayed ops, so we do that first.
     return this.drain().then(() => {
       // Run ops in the same order they'd run if they ran naturally.
+      /* eslint-disable @typescript-eslint/no-floating-promises */
```
Comment (DellaBitta) [inline packages/firestore/src/util/async_queue_impl.ts:289]: here, too

File: packages/firestore/src/util/async_queue_impl.ts [hunk 4]
```diff
@@ -310,6 +311,7 @@ export class AsyncQueueImpl implements AsyncQueue {
   private removeDelayedOperation(op: DelayedOperation<unknown>): void {
     // NOTE: indexOf / slice are O(n), but delayedOperations is expected to be small.
     const index = this.delayedOperations.indexOf(op);
+    /* eslint-disable @typescript-eslint/no-floating-promises */
```
Comment (DellaBitta) [inline packages/firestore/src/util/async_queue_impl.ts:314]: here, too

File: packages/analytics/src/helpers.ts (hunk 1 above)
Comment (dlarocque) [reply packages/analytics/src/helpers.ts]: I replaced all instances of this with `eslint-disable-next-line` which has the same effect in a single comment.
I forgot `eslint-disable` was for entire files- thanks for reminding me!

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**